### ISSUE TITLE
Pulling planet further than intended.

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/keplers-laws#169]
**Test device**
Samsung
**Operating System**
Win 11
**Browser**
Chrome
**Problem description**
For https://github.com/phetsims/qa/issues/979
It is possible to drag the planet further that it seems was intended. Combined with changing the window size this can lead to the planet being off screen.
**Steps to reproduce**
1. Drag the planet to the right side of the screen, under any toolbox
2. Even after it seems to hit an edge, continue pulling to the side, moving mouse back and forth
3. Using keyboard nav makes this even easier.
4. This seems to stop at the edge of the screen, but changing the window size to something smaller can put the planet outside the bounds.

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Kepler's Laws‬
URL: https://phet-dev.colorado.edu/html/keplers-laws/1.1.0-dev.5/phet/keplers-laws_all_phet.html
Version: 1.1.0-dev.5 2023-09-07 23:32:34 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36
Language: en-US
Window: 1536x707
Pixel Ratio: 1.25/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 4096
Texture: size: 8192 imageUnits: 32 (vertex: 32, combined: 64)
Max viewport: 8192x8192
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by AgustinVallejo** [comment]
Could you maybe record the screen to see what you're seeing? In my end it is working as intended, i.e. the drag bounds are defined by the screen and the panels (see image below).

![image](https://github.com/phetsims/keplers-laws/assets/37052186/be4367b8-0439-4ad6-95b1-5f64692aae6c)


As for changing screen size and having the planet outside, I'm not sure if we have discussed that scenario or if anything should be done about it. @DianaTavares what do you think?

**Comment by DianaTavares** [comment]
I think that the simulation needs to do the same that Density. If the user drags the planet to the edge of the screen, and later makes the screen smaller, the edge of the screen needs to push the planet to always have it visible.

This is the actual behavior. 

1) Planet raged to the edge of the panel:
![image](https://github.com/phetsims/keplers-laws/assets/26031501/b8912445-8d47-4aae-8c0f-f83436a2d2f1)

2) screen smaller, then, the planet is behind the panel:
![image](https://github.com/phetsims/keplers-laws/assets/26031501/8104c14a-2286-4eb2-a84f-867ffa208ffd)

Modify it to make that the edge push the planet to the play area (that is going to modify the orbit). 

Also, the planet can be dragged behind the zoom controls and the play. Can you modify it to make it as the reset button?

**Comment by KatieWoe** [comment]
As I'm pulling it to the side you can see it get "stuck" before the edge of the play area. Moving it back and forth gets it closer and closer to the edge.

https://github.com/phetsims/keplers-laws/assets/41024075/faf3cfa5-1fb6-4128-b803-d8617a06fe55

**Comment by AgustinVallejo** [comment]
@DianaTavares I think pushing the planet inwards is not a feature we want, as it'll change the orbit as you said, and it could be confusing for users. Similarly, if we go that route, we would have to push bodies inside in My Solar System as well, and I think it would potentially affect user made systems. What do you think?

**Comment by DianaTavares** [comment]
You are right. Then is better to leave the behavior just the way it is. The user doesn't lose it forever. They can always click play and let the planet move to a visible area, or refresh the simulation. 
Changing the orbit should always be a a direct action of a manipulation of the relevant variables by the user (consciously changing the position or speed of the planet) and not a rearrangement of the size of the window.

**Comment by AgustinVallejo** [comment]
And regarding the speed change reported by @KatieWoe, it seems a major refactoring would be necessary to fix this, and the bug is not really that bad. It's caused by an unfortunate arrange in the drag bounds logic:
1. The body checks that it's distance is not bigger than the escape radius (If it gets further, it would escape with the current velocity). If it is, set the distance to be just under the escape distance.
2. However, if dragged over a panel, the function `constrainBoundaryViewPoint` will look for the closest point within the drag bounds and set the distance to that. 

Occasionally, step 2 overrides the limit of step 1, thus placing the body further than escape distance. Fortunately, the model immediately detects the body would escape and reduces the current speed for it to be just under the escape speed.

If we wanted to fix this, step 2 would have to be enhanced for it to check if the new distance respects step 1, but because  `constrainBoundaryViewPoint` is written for a very general case, I don't think it's worth the trouble of specifying what the escape distance is.

In any case, this bug is not sim breaking nor affects the experience of the user. Other situations of the sim will also change the speed to respect the escaping limit, such as reducing the mass of the sun. Marking this as deferred.
